src/model/version.py

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2022 spdx contributors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import re
+from re import Pattern
+
+
+class Version:
```

Comment:
Imho, the main use case would be something like "has same major version". As soon as we need that, I'd prefer to have a proper class for the version instead of doing regex matching every time.
That said, it doesn't look like we have such use cases so far. Part of the reason for carrying it over was because it was there in the previous data model. I'm fairly indifferent whether we keep a proper class or a plain string, with slight preference for the class since it's already there and makes it clearer what we expect of the version.